infra/infra-queue/infra-queue-consumer/src/main/java/com/sflpro/notifier/queue/consumer/notification/config/KafkaConsumerConfiguration.java

```diff
@@ -26,32 +37,90 @@
 @EnableKafka
 public class KafkaConsumerConfiguration {
 
+    private static final Logger logger = LoggerFactory.getLogger(KafkaConsumerConfiguration.class);
+
     @Value("${kafka.bootstrap.servers}")
     private String bootstrapServers;
 
+    @Value("${kafka.heartbeat.interval.ms}")
+    private Integer heartbeatInterval;
+
+    @Value("${kafka.concurrency.factor}")
+    private int kafkaConcurrencyFactor;
+
+    @PostConstruct
+    public void validateConfig() {
+        if (kafkaConcurrencyFactor < 1) {
+            throw new IllegalStateException("Kafka concurrency factor should be a positive integer");
+        }
+    }
+
     @Bean
     public Map<String, Object> setupConsumerConfig() {
         Map<String, Object> props = new HashMap<>();
         props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+        props.put(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, heartbeatInterval);
         props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
         props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
-        props.put(ConsumerConfig.GROUP_ID_CONFIG, "json");
+        props.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1);
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, "notifier-worker");
 
         return props;
     }
 
     @Bean
     public ConsumerFactory<String, byte[]> setupConsumerFactory() {
-        return new DefaultKafkaConsumerFactory<>(setupConsumerConfig(), new StringDeserializer(),
+        ConsumerFactory<String, byte[]> consumerFactory = new DefaultKafkaConsumerFactory<>(setupConsumerConfig(), new StringDeserializer(),
                 new ByteArrayDeserializer());
+        logger.info("Kafka consumer factory initialized");
+        return consumerFactory;
     }
 
     @Bean
     public ConcurrentKafkaListenerContainerFactory<String, byte[]> kafkaListenerContainerFactory() {
         ConcurrentKafkaListenerContainerFactory<String, byte[]> factory =
                 new ConcurrentKafkaListenerContainerFactory<>();
         factory.setConsumerFactory(setupConsumerFactory());
+        factory.setRetryTemplate(retryTemplate());
+        factory.setConcurrency(kafkaConcurrencyFactor);
+        factory.getContainerProperties().setAckMode(AbstractMessageListenerContainer.AckMode.MANUAL);
+        factory.getContainerProperties().setAckOnError(false);
+        /*
+        try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)